Mathlib/Data/Finset/Basic.lean

```diff
@@ -799,14 +799,33 @@ theorem eq_empty_of_ssubset_singleton {s : Finset α} {x : α} (hs : s ⊂ {x})
   ssubset_singleton_iff.1 hs
 #align finset.eq_empty_of_ssubset_singleton Finset.eq_empty_of_ssubset_singleton
 
-theorem eq_singleton_or_nontrivial (ha : a ∈ s) : s = {a} ∨ (s : Set α).Nontrivial := by
-  rw [← coe_eq_singleton]
-  exact Set.eq_singleton_or_nontrivial ha
+/-- A finset is nontrivial if it has at least two elements. -/
+@[reducible]
+protected def Nontrivial (s : Finset α) : Prop := (s : Set α).Nontrivial
+#align finset.nontrivial Finset.Nontrivial
+
+@[simp]
+theorem not_nontrivial_empty : ¬ (∅ : Finset α).Nontrivial := by simp [Finset.Nontrivial]
+#align finset.not_nontrivial_empty Finset.not_nontrivial_empty
+
+@[simp]
+theorem not_nontrivial_singleton : ¬ ({a} : Finset α).Nontrivial := by simp [Finset.Nontrivial]
+#align finset.not_nontrivial_singleton Finset.not_nontrivial_singleton
+
+theorem Nontrivial.ne_singleton (hs : s.Nontrivial) : s ≠ {a} := by
+  rintro rfl; exact not_nontrivial_singleton hs
+#align finset.nontrivial.ne_singleton Finset.Nontrivial.ne_singleton
+
+theorem eq_singleton_or_nontrivial (ha : a ∈ s) : s = {a} ∨ s.Nontrivial := by
+  rw [← coe_eq_singleton]; exact Set.eq_singleton_or_nontrivial ha
 #align finset.eq_singleton_or_nontrivial Finset.eq_singleton_or_nontrivial
 
-theorem Nonempty.exists_eq_singleton_or_nontrivial :
-    s.Nonempty → (∃ a, s = {a}) ∨ (s : Set α).Nontrivial := fun ⟨a, ha⟩ =>
-  (eq_singleton_or_nontrivial ha).imp_left <| Exists.intro a
+theorem nontrivial_iff_ne_singleton (ha : a ∈ s) : s.Nontrivial ↔ s ≠ {a} :=
+  ⟨Nontrivial.ne_singleton, (eq_singleton_or_nontrivial ha).resolve_left⟩
+#align finset.nontrivial_iff_ne_singleton Finset.nontrivial_iff_ne_singleton
+
+theorem Nonempty.exists_eq_singleton_or_nontrivial : s.Nonempty → (∃ a, s = {a}) ∨ s.Nontrivial :=
+  fun ⟨a, ha⟩ => (eq_singleton_or_nontrivial ha).imp_left <| Exists.intro a
 #align finset.nonempty.exists_eq_singleton_or_nontrivial Finset.Nonempty.exists_eq_singleton_or_nontrivial
 
 instance [Nonempty α] : Nontrivial (Finset α) :=
```

Comment:
Can you add the #align here? I agree there's no need to prime the name any more.